### DISCUSSION TITLE
Ecommerce session vs database carts

### DISCUSSION THREAD

**Discussion by alasdair-shields** [pimcore/pimcore#17115] [Q&A / Support] [answered]
What is the recommended cart startegy when you want users to be able to add items to the cart without logging in but need them to be logged in when making payment.  Currently I'm using the session cart then transferring items to a new db cart on log-in but I was wondering if there was a nicer way to do it.  

I tried just using the session cart but can't find any way to do the post-payment processing when the payment provider responds (as this request is not part of the user session the sesion cart is unavailable).  Any suggestions?

**Comment by markus-moser** [comment, marked as answer]
Yes, using the session cart when not logged in and transfering the items to the DB cart on login is the best strategy. Unfortunately there is no default service/option for this as it depends on the use case, so you need to do this in your custom code.

Regarding the payment topic: It is better to do the payment handling based on order objects instead of directly doing it at the cart level. The ecommerce framework offers different strategies for dealing with pending payments (so for the timeframe between initializing the payment and receiving the final response from the payment provider):

https://pimcore.com/docs/platform/Ecommerce_Framework/Checkout_Manager/Integrating_Payment/#dealing-with-pending-payments

I would vote for the usage of the `AlwaysRecreateOrderStrategy`.

**Reply by alasdair-shields** [reply]
Thanks for your quick response. I'll continue using my curent cart strategy.  

On payment I just needed the cart to clear out the paid-for items at the "// ... do some stuff, and get $cart" step here https://pimcore.com/docs/pimcore/10.6/Development_Documentation/E-Commerce_Framework/Checkout_Manager/Integrating_Payment.html (still on v10.6 - wish they'd slow down with the releases to let me catch up :-)).
